block_pickup.php

```diff
@@ -152,11 +152,16 @@ public function fetch_recent_courses(): array {
             $course->coursecategory = $cr->catname;
 
             /* Progress. */
-            if ($percentage = progress::get_course_progress_percentage($cr, $USER->id)) {
-                $percentage = floor($percentage);
-                $course->progress = $percentage;
+            if ($cr->enablecompletion) {
+                $course->progressenabled = true;
+                $course->progress = "0"; // Default.
```

Comment:
Why is this "0" (string) instead of:

                $course->progress = 0;

Two lines below `$course->progress` is assigned the result of `floor()` which is numeric albeit a float.